src/backend/distributed/commands/alter_table.c

```diff
@@ -1122,9 +1125,27 @@ ReplaceTable(Oid sourceId, Oid targetId, List *justBeforeDropCommands,
 									quote_qualified_identifier(schemaName, sourceName))));
 		}
 
-		appendStringInfo(query, "INSERT INTO %s SELECT * FROM %s",
-						 quote_qualified_identifier(schemaName, targetName),
-						 quote_qualified_identifier(schemaName, sourceName));
+		if (HasAnyGeneratedStoredColumns(sourceId))
+		{
+			/* see comment in BuildInsertColumnStringForReplaceTable */
+			char *insertColumnString = BuildInsertColumnStringForReplaceTable(sourceId);
+			appendStringInfo(query, "INSERT INTO %s (%s) SELECT %s FROM %s",
+							 quote_qualified_identifier(schemaName, targetName),
+							 insertColumnString, insertColumnString,
+							 quote_qualified_identifier(schemaName, sourceName));
+		}
+		else
+		{
+			/*
```

Comment:
Minor: I think this comment is good, but it felt that we are not reflecting the vast majority of the cases: Table has columns :) 

So, maybe start with that in a single sentence.